tools/xcms/xcms_plot_raw.xml

```diff
@@ -0,0 +1,57 @@
+<tool id="xcms_plot_raw" name = "Plot filtered by m/z and retention time" version="@TOOL_VERSION@+galaxy0" profile="21.09">
+    <description>Plot raw data filtered by m/z range and retention time (RT) range</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="creator"/>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">bioconductor-xcms</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        Rscript -e 'source("${plot_raw}")'
+    ]]></command>
+    <configfiles>
+        <configfile name="plot_raw">
+library(xcms)
+library(MsExperiment)
+library(Spectra)
+mse = readMsExperiment(file.path('${input}'))
+mz_offset = ${tolerance_ppm} * 1e-6 * ${mz_value}
+rt_offset = ${rt_range} / 2
+raw = mse |>
+    filterRt(rt = ${rt} + c(-rt_offset, rt_offset)) |>
+    filterMzRange(mz = ${mz_value} + c(-mz_offset, mz_offset))
+png(filename = '${output_filename}')
+par(oma = c(0.5, 2, 0.5, 1))
+plot(raw)
+dev.off()
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param type="data" name="input" format="mzml" label="Input mzML file"/>
+        <param type="float" name="mz_value" value="0.0" label="m/z Value" help="m/z value for the plot"/>
+        <param type="integer" name="tolerance_ppm" value="10" label="Tolerance (ppm)" help="Tolerance for m/z value in ppm"/>
+        <param type="float" name="rt" label="Retention Time" value="0.0" help="Retention time for the plot"/>
+        <param type="float" name="rt_range" value="5.0" label="Retention Time Range" help="Retention time range for the plot"/>
```

Comment:
please add reasonable minimum values to these parameters

Reply:
Added -> [3944629](https://github.com/RECETOX/galaxytools/pull/523/commits/39446294577066fc66e03ec19dce25f86c645aed)